### PR TITLE
add presubmit job for gardener azure integration tests

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- consolidate all gardener integration tests in one single template
- move gardener azure from periodic to presubmit job (optional, as it is flaky)

**Related issue(s)**

### CODE REVIEW

File: templates/config.yaml [hunk 1]
```diff
@@ -47,6 +48,7 @@ kubebuilder2_buildpack_v20200124: &kubebuilder2_buildpack_v20200124 eu.gcr.io/ky
 bootstrap_buildpack_v20181121: &bootstrap_buildpack_v20181121 eu.gcr.io/kyma-project/prow/test-infra/bootstrap:v20181121-f3ea5ce
 bootstrap_buildpack_v20181204: &bootstrap_buildpack_v20181204 eu.gcr.io/kyma-project/prow/test-infra/bootstrap:v20181204-a6e79be
 
+
```
Comment (marcobebway) [inline templates/config.yaml:51]: (nit) Unnecessary change.

File: templates/config.yaml [hunk 2]
```diff
@@ -38,6 +38,7 @@ global:
       project: "kyma-incubator"
       base_ref: "kyma-master"
 
+
```
Comment (marcobebway) [inline templates/config.yaml:41]: (nit) Unnecessary change.

File: prow/jobs/kyma/kyma-integration-gardener.yaml [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+#        - <<: *test_infra_ref
+#          base_ref: master
+
+periodics:
+  - name: kyma-integration-gardener-gcp
+    <<: *gardener_gcp_job_template
+    decorate: true
+    decoration_config:
+      timeout: 14400000000000 # 4h
+      grace_period: 600000000000 # 10min
```
Comment (marcobebway) [inline prow/jobs/kyma/kyma-integration-gardener.yaml:154]: (nit) Could these values be like:
```
timeout: 4h
grace_period: 10m
```

If yes, then please apply to the rest of the template.

File: templates/config.yaml [hunk 4]
```diff
@@ -564,6 +566,16 @@ templates:
...
+  - from: templates/kyma-integration-gardener.yaml
+    render:
+      - to: ../prow/jobs/kyma/kyma-integration-gardener.yaml
+        values:
+          optional: true
+          resource_changes_triggering_kafka_tests:
+            - resources/knative-eventing
+            - resources/knative-eventing-kafka
+            - resources/knative-serving
+            - resources/application-connector
```
[the remark this answers is not in the document]
Comment (k15r) [reply templates/config.yaml:578]: all subfolders are also included

[the remark this answers is not in the document]
Comment (marcobebway) [reply templates/config.yaml:578]: Yes, the question is do we need to trigger the job when all other sub-directories change.

File: templates/templates/kyma-integration-gardener.yaml [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+#        - <<: *test_infra_ref
+#          base_ref: master
+
+periodics:
+  - name: kyma-integration-gardener-gcp
+    <<: *gardener_gcp_job_template
+    decorate: true
+    decoration_config:
+      timeout: 14400000000000 # 4h
+      grace_period: 600000000000 # 10min
```
[the remark this answers is not in the document]
Comment (k15r) [reply templates/templates/kyma-integration-gardener.yaml:152]: > A Duration represents the elapsed time between two instants as an int64 nanosecond count. The representation limits the largest representable duration to approximately 290 years.
In short: nope, not possible

File: templates/config.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (k15r) [reply templates/config.yaml:576]: the top folder contains the values.yaml, which contains all image tags. so we need to definitely include the topfolder

File: templates/templates/kyma-integration-gardener.yaml (hunk 5 above)
[the remark this answers is not in the document]
Comment (k15r) [reply templates/templates/kyma-integration-gardener.yaml:152]: time.Duration is int64 ... Nanoseconds between two timestamps.
